pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -381,6 +384,24 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
 			pvcSource.ClaimName,
 			pvcUID)
 
+		// TODO: remove feature gate check after no longer needed
+		if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+			volumeMode, err := volumehelper.GetVolumeMode(volumeSpec)
+			if err != nil {
+				return nil, "", err
+			}
+			// Error if a container has volumeMounts but the volumeMode of PVC isn't Filesystem
+			// or if a container has volumeDevices but the volumeMode of PVC isn't Block
+			if (mountsMap[podVolume.Name] && volumeMode != v1.PersistentVolumeFilesystem) ||
```

Comment:
Can you split the checks and error into two?  Then the error will be more specific.

Reply:
ack.